### PR TITLE
Include node status in kubectl get output

### PR DESCRIPTION
**Pull request by deyuanix**

### CODE REVIEW

**Comment by ddysher** [comment]
@bgrant0607 @lavalamp

File: pkg/kubectl/resource_printer.go [hunk 1]
```diff
@@ -347,7 +347,30 @@ func printServiceList(list *api.ServiceList, w io.Writer) error {
...
-	_, err := fmt.Fprintf(w, "%s\t%s\n", minion.Name, formatLabels(minion.Labels))
+	var status []string
+	conditionMap := make(map[api.NodeConditionKind]*api.NodeCondition)
+	for i := range minion.Status.Conditions {
+		cond := minion.Status.Conditions[i]
+		conditionMap[cond.Kind] = &cond
+	}
+	if condition, ok := conditionMap[api.NodeReady]; ok {
+		if condition.Status == api.ConditionFull {
+			status = append(status, "Ready")
```
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: I'm fine with this for now, but when we add more conditions, we should make this generic: strip off the Node prefix and conditionally prepend Not to the condition kind. Add a TODO (or fix it now, if you prefer).

File: pkg/kubectl/resource_printer.go [hunk 2]
```diff
@@ -347,7 +347,30 @@ func printServiceList(list *api.ServiceList, w io.Writer) error {
...
+	}
+	if condition, ok := conditionMap[api.NodeReachable]; ok {
+		if condition.Status == api.ConditionFull {
+			status = append(status, "Reachable")
+		} else {
+			status = append(status, "NotReachable")
+		}
+	}
+	if len(status) == 0 {
+		status = append(status, "None")
```
Comment (bgrant0607) [inline pkg/kubectl/resource_printer.go]: Unknown, for consistency with phase.

File: pkg/kubectl/resource_printer.go (hunk 1 above)
Comment (ddysher) [inline pkg/kubectl/resource_printer.go]: Fixed, no need to defer ;)

**Comment by ddysher** [comment]
Comment addressed.  PTAL

**Comment by dchen1107** [comment]
I noticed that node status is not cached in etcd store. In this case, we might end up to HTTP health check against all nodes when running kubectl get nodes, right? 

I am ok with it for now.

**Comment by ddysher** [comment]
Yes, this is already the case now.  We will cache the status for sure, it's not scalable.

**Pull request merged**